facturacion_mexico/cfdi_recibidos/tests/test_cfdi_recibido_parser.py

```
@@ -0,0 +1,279 @@
+		self.assertIn("002", impuestos)  # IVA retenido
+
+	def test_concepto_retenciones(self):
+		c = self.data["conceptos"][0]
+		taxes = json.loads(c["taxes_json"])
+		self.assertEqual(len(taxes["retenciones"]), 2)
+
+
+class TestCFDIRecibidoParserVersionInvalida(unittest.TestCase):
+	def test_rechaza_cfdi_33(self):
+		"""CFDI 3.3 debe lanzar ValidationError."""
+		with self.assertRaises(Exception) as ctx:
```

Ruff (0.15.13)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)